src/components/CourseContainer.tsx

```diff
@@ -0,0 +1,55 @@
+import LessonsList from "./LessonsList";
+import LessonContainer from "./LessonContainer";
+import { CourseDetails } from "../models/courseDetailsModel";
+import { useLocalStorage } from "../hooks/useLocalStorage";
+import { useEffect } from "react";
+import { useAppDispatch, useAppSelector } from "../redux/store";
+import { setVideosProgressStorage } from "../redux/slices/videosProgressSlice";
+
+type Props = {
+	course: CourseDetails;
+};
+
+export default function CourseContainer(props: Props) {
+	const { course } = props;
+	const dispatch = useAppDispatch();
+	const { videosProgresStorage } = useAppSelector(
+		(state) => state.videosProgress
+	);
+	const [videosProgressStorageLS, setVideosProgressStorageLS] =
+		useLocalStorage<Record<string, number> | null>(
+			"videos-progress-storage",
+			null,
+			3600
+		);
+	const firstLesson = course.lessons[0];
+
+	useEffect(() => {
+		if (videosProgressStorageLS == null) return;
+		dispatch(setVideosProgressStorage(videosProgressStorageLS));
+	}, []);
+
+	useEffect(() => {
+		if (videosProgresStorage == null) return;
```

Comment:
There is a reason why I use double-equal here instead of strict equality: if for some reason we will get `undefined` as a value from local storage this will prevent an error as long as we know that `null == undefined`.

Reply:
If you want to check to falsy values better use  `!videosProgresStorage`, it's more obvious then == null